### ISSUE TITLE
I am trying to register user as foreign key in administrator. But when I start runserver command I am getting this issue:Attribute error:  'Post object has no attribute 'is_anonymous'

### ISSUE THREAD

**Issue by mayur200** [mayur200/django3#1]
I was referring this [django 3](https://docs.djangoproject.com/en/3.1/topics/auth/customizing/#referencing-the-user-model) docs to create user as ForeignKey 
[Click Here for doc](https://docs.djangoproject.com/en/3.1/topics/auth/customizing/#referencing-the-user-model)

> My file path in git is : https://github.com/mayur200/django3/blob/master/my_blog/sign_up/models.py

I have written code in models.py file
**models.py**

```
from django.db import models
from django.contrib.auth.models import  User
from django.conf import settings
from django.db.models.signals import post_save
from django.contrib.auth.models import AbstractUser

class User(AbstractUser):
    pass

class Post(models.Model):
    
    REQUIRED_FIELDS = ('user',)


    post = models.CharField(max_length=500)
    USERNAME_FIELD = 'post'

    # upostser = models.ForeignKey(User, on_delete=models.CASCADE,  default='none')
    user = models.ForeignKey(settings.AUTH_USER_MODEL,  on_delete=models.CASCADE,)


    def post_save_receiver(sender, instance, created, **kwargs):
        pass

    post_save.connect(post_save_receiver, sender=settings.AUTH_USER_MODEL)

```

**And my error traceback is:**

_File "/usr/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/opt/env/venv/lib/python3.6/site-packages/django/utils/autoreload.py", line 53, in wrapper
    fn(*args, **kwargs)
  File "/opt/env/venv/lib/python3.6/site-packages/django/core/management/commands/runserver.py", line 117, in inner_run
    self.check(display_num_errors=True)
  File "/opt/env/venv/lib/python3.6/site-packages/django/core/management/base.py", line 395, in check
    include_deployment_checks=include_deployment_checks,
  File "/opt/env/venv/lib/python3.6/site-packages/django/core/management/base.py", line 382, in _run_checks
    return checks.run_checks(**kwargs)
  File "/opt/env/venv/lib/python3.6/site-packages/django/core/checks/registry.py", line 72, in run_checks
    new_errors = check(app_configs=app_configs)
  File "/opt/env/venv/lib/python3.6/site-packages/django/contrib/auth/checks.py", line 74, in check_user_model
    if isinstance(cls().is_anonymous, MethodType):
**AttributeError: 'Post' object has no attribute 'is_anonymous'**_

**Comment by ksmeeks0001** [comment]
Looks like user method is_anonymous is being called on Post when it should be post.user.is_anonymous

**Comment by mayur200** [comment]
> Looks like user method is_anonymous is being called on Post when it should be post.user.is_anonymous

Actually I have called Post objet in **views.py** file but there I tried _post.user.is_anonymous_ but still  I am gettin same issue

 **I have tried this:**
**views,py**
def post_blog(request):
    global text, form
    rating = 'Error 404'
    context = {'form': rating}

    # t=0
    if request.method == 'POST':
        form = HomeForm(request.POST)
        if form.is_valid():
            post = form.save(commit=False)
            post.user = request.user
            post.user.is_anonymous
            post.save
            text = form.cleaned_data["post"]
            form = HomeForm()
            # return redirect('blog')
        else:
            text = 'error'

        args = {'form':form,'text':text}
        return  render(request, 'testapp/home.html', args)
    else:
        return render(request, 'testapp/home.html', context)

**Comment by ksmeeks0001** [comment]
post.user.is_anonymous() should be called with parentheses. It returns a boolean value so it's not doing anything in your code. If you're trying to check of user is logged in or not use 
if post.user.is_anonymous():
    something...

**Comment by mayur200** [comment]
> post.user.is_anonymous() should be called with parentheses. It returns a boolean value so it's not doing anything in your code. If you're trying to check of user is logged in or not use
> if post.user.is_anonymous():
> something...

Actually I have debugged it but while using runserver command 
 `def post_blog(request): ` method does not get called. But still I have tried` if post.user.is_anonymous():` still I am getting same error. Can you give any other solution?

**Comment by mayur200** [comment]
I have solved it. This issue was raised because I was working on Django 3.4.0 version.
1)keep django version 2.2.0 (the issue addressed in this version) `pip install django==2.2.0`
2)Delete the SQLite db
3)Migrate again python manage.py makemigrations and then
4) `python manage.py migrate`
5)Start the server `python manage.py runserver`
